man/patterns.Rd

```diff
@@ -16,18 +16,25 @@ patterns(\dots, cols=character(0))
 }
 \arguments{
   \item{\dots}{A set of regular expression patterns.}
-  \item{cols}{A character vector of names to which each pattern is matched.}
+  \item{cols}{A character vector of column names to match with
+    patterns. When used in the context of
+    \code{melt(measure.vars=patterns())} or
+    \code{DT[, .SDcols=patterns()]}, the
+    user should not provide the \code{cols} argument, which is
+    automatically set to all of the column names of the input data table.
+  }
 }
 \seealso{
-  \code{\link{melt}},
-  \url{https://github.com/Rdatatable/data.table/wiki/Getting-started}
+  \code{\link{melt}}, \code{\link{.SD}},
+  \url{https://github.com/Rdatatable/data.table/wiki/Getting-started},
+  \url{https://cloud.r-project.org/web/packages/data.table/vignettes/datatable-reshape.html},
```

Comment:
nit: possibly preferable to use `vignette("datatable-reshape")`? that way users reading this page from an airgapped machine can still access the resource.

Or perhaps, whatever the Rd equivalent of markdown:

```markdown
[`vignette("datatable-reshape")`](https://cloud.r-project.org/web/packages/data.table/vignettes/datatable-reshape.html)
```

would be to get the best of both worlds (link the HTML version where links are available, but show the code to access from anywhere)?